### PR TITLE
Feat 4471 deepgram language detection function using java

### PR DESCRIPTION
**Pull request by Ambareen09**
fixes https://github.com/appwrite/appwrite/issues/4471

language

### CODE REVIEW

File: java/deepgramLanguageDetection/Index.java [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        Map<String, Object> responseData=new HashMap<>();
+
+        try{
+            deepgramData=detectLanguage(fileurl,apiKey);
+        }catch(Exception e){
+            responseData.put("success",false);
+            responseData.put("message","Something went wrong while generating the summary, please check with the developers. Error: "+e.getMessage());//TODO
+        return res.json(responseData);
+        }
+        responseData.put("success",true);
```
Comment (adityaoberai) [inline java/deepgramLanguageDetection/Index.java:55]: Can we add a check for the status code within the Deepgram API's request and send a success/failure response accordingly?

In its current state, the function might return a successful response even if Deepgram's API returns a Bad Request or Unauthorized error.

**Review by adityaoberai — CHANGES_REQUESTED** [review]
Thanks for the PR @Ambareen09!
Please can you make the suggested changes and ping us here?

File: java/deepgramLanguageDetection/Index.java (hunk 1 above)
Comment (Ambareen09) [reply java/deepgramLanguageDetection/Index.java:55]: 400s error gets handled in try catch,
Screenshot from 2022-10-29 22-33-43

File: java/deepgramLanguageDetection/Index.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        HttpURLConnection con=(HttpURLConnection)url.openConnection();
+
+        con.setRequestMethod("POST");
+        con.setRequestProperty("Content-Type","application/json");
+        con.setRequestProperty("Authorization","Token "+apiKey);
+
+        con.setDoOutput(true);
+
+        //prepare request body
+        requestBody="{"url":"requestBody"}";
```
Comment (adityaoberai) [inline java/deepgramLanguageDetection/Index.java:83]:
```suggestion
        requestBody="{\"url\":\"requestBody\"}";
```

The escape sequence is necessary to add `"` inside a string

File: java/deepgramLanguageDetection/Index.java [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+        OutputStream os=con.getOutputStream();
+        byte[] input=requestBody.getBytes("utf-8");
+        os.write(input,0,input.length);
+
+        StringBuilder response=new StringBuilder();
+
+        BufferedReader br=new BufferedReader(new InputStreamReader(con.getInputStream(),"utf-8"));
+        String responseLine=null;
+        responseData.put("success",false);
```
Comment (adityaoberai) [inline java/deepgramLanguageDetection/Index.java:93]:
```suggestion
```

This line is unnecessary

**Review by adityaoberai — CHANGES_REQUESTED** [review]
Need to make a few more changes

File: java/deepgramLanguageDetection/Index.java (hunk 2 above)
Comment (Ambareen09) [reply java/deepgramLanguageDetection/Index.java]: My curl request has stopped working, does this change require a change in current structure of payload?
```bash
curl --location --request POST 'http://localhost:3000/' \
--header 'X-Internal-Challenge: secret-key' \
--header 'Content-Type: application/json' \
--data-raw '{"payload": "{\"fileUrl\": \"https://static.deepgram.com/examples/interview_speech-analytics.wav\"}", 
"variables": {"DEEPGRAM_SECRET_KEY":"<YOUR_API_KEY>"}}
'
```

**Comment by Ambareen09** [comment]
@adityaoberai, Can you please review it?

**Review by adityaoberai — APPROVED** [review]
PR looks good to me 👍

**Comment by Ambareen09** [comment]
Can you please add `hacktoberfest-accepted` label if you're fine with this PR?

**Comment by christyjacob4** [comment]
THANK YOU! All changes merged 🥳

Please reach out to me on our [Discord server](https://appwrite.io/discord) if you would like to claim your Appwrite swags! As a way of saying thank you, we would also love to invite you to join the Appwrite organization on GitHub. Please share your GitHub username with us on Discord.  

You can accept the invite by visiting https://github.com/orgs/appwrite/invitation. By joining our team, you will officially be an Appwrite maintainer on GitHub.

You can change your membership [visibility settings](https://help.github.com/en/github/setting-up-and-managing-your-github-user-account/publicizing-or-hiding-organization-membership), so your new Appwrite team membership badge will show up on your personal GitHub profile.

Please feel free to look for more PRs you might be interested in helping with on our long list of [Hacktoberfest friendly issues](https://github.com/search?q=org%3Aappwrite+org%3Autopia-php+org%3Aopen-runtimes+label%3Ahacktoberfest+created%3A%3E2021-01-01+state%3Aopen&type=issues&s=updated&o=asc) and help make Appwrite better :)

**Pull request merged**